src/t_stream.c

```diff
@@ -400,50 +400,31 @@ int streamCompareID(streamID *a, streamID *b) {
 }
 
 /* Retrieves the ID of the stream edge entry. An edge is either the first or
- * the last ID in the stream. */
-void streamGetEdgeID(stream *s, int first, streamID *edge_id)
+ * the last ID in the stream, and may be a tombstone. To filter out tombstones,
+ * set the'skip_tombstones' argument to 1. */
+void streamGetEdgeID(stream *s, int first, int skip_tombstones, streamID *edge_id)
 {
-    raxIterator ri;
-    raxStart(&ri, s->rax);
-    int empty;
-    if (first) {
-        raxSeek(&ri, "^", NULL, 0);
-        empty = !raxNext(&ri);
-    } else {
-        raxSeek(&ri, "$", NULL, 0);
-        empty = !raxPrev(&ri);
+    streamIterator si;
+    int64_t numfields;
+    int empty = (s->length == 0);
+    streamID min_id, max_id;
+    min_id.ms = 0;
+    min_id.seq = 0;
+    max_id.ms = UINT64_MAX;
+    max_id.seq = UINT64_MAX;
+
+    if (!(skip_tombstones && empty)) {
```

Comment:
don't we need to use the streamIteratorStart anyway? 
i.e. 
```suggestion
    if (!empty) {
```

Reply:
I don't think so - if we don't want tombstones included, and the stream's length is 0 then we don't need to iterate and can jump to min/max response.

Reply:
Ohh, ok. i think this form is easier to understand, but also let's add a comment (both to explain what that block does, and also the condition)
```suggestion
    /* If the stream is not empty, locate the edge ID (The stream may look empty, but might still have tombstones). */
    if (!empty || !skip_tombstones) {
```